### DISCUSSION TITLE
what would be the best way to install the existing database from the .sql file?

### DISCUSSION THREAD

**Discussion by infrared-o8** [infrared-o8/hospital-management-python-sql#6] [Q&A] [answered]
im seeing several problems with the .bat file. what I'm thinking now, is to add a separate python file that executes each command in the .SQL file. please give me suggestions 🙏 

the bat file will be reconfigured to install the requirements.txt modules and run the new python script.

**Comment by Mohammad-Hamzah** [comment]
i actually had the python code for this
wait lemme see

**Comment by Mohammad-Hamzah** [comment, marked as answer]
```

def execute_sql_file(userinp, userpassword):

    try:
        database.database='hospital_main'
    except:
        c.execute("CREATE DATABASE hospital_main")

        mysql_executable = r'C:\Program Files\MySQL\MySQL Server 8.0\bin\mysql.exe'
        command = [mysql_executable, "-u", userinp, "-p" + userpassword, 'hospital_main']

        with open(r'hospital_main.sql', 'r') as sql_file:
            subprocess.run(command, stdin=sql_file)

        database.database='hospital_main'

execute_sql_file(userinp,userpassword)

```

this should do the job :)
**imp point: in your initial cursor declaration database shouldnt be declared...
also dont forger to import subprocess**

**Comment by Mohammad-Hamzah** [comment]
this should be it for requirements
**dont forget to import sys**

```
flag_file = '.first_run'

# Check if the flag file exists
if not os.path.exists(flag_file):
    subprocess.run([sys.executable, "-m", "pip", "install", "-r", "requirements.txt"])
    with open(flag_file, 'w') as f:
        f.write("This file marks the first run completion.")
```

😄

**Comment by Mohammad-Hamzah** [comment]
`subprocess.run(["attrib","+H",flag_file])`

use this to hide the file if you want
also i think these commands wont raise an error unless you put the check=True thingy at the end...
for eg. here:
`subprocess.run(["attrib","+H",flag_file],check=True)`

**Comment by infrared-o8** [comment]
aight i have implemented your code. thnkuuu 😭 🙏 

ill test it on the PC in our class today. 

the code should be in install_database.py

**Comment by Mohammad-Hamzah** [comment]
okii.... hope it works ❤️ 😭
